cdp_scrapers/instances/houston.py

```diff
@@ -0,0 +1,289 @@
+        return element
+
+    def get_body_name(self, event: Union[Tag, NavigableString, None]) -> str:
+        """
+        Get the body name for an event
+
+        Parameter:
+        ----------------
+        event: Union[Tag, NavigableString, None]
+            All elements in the page that we want to scrape
+
+        Returns:
+        ----------------
+        str
+            The body name
+        """
+        log.info("start get body name")
+        event = self.remove_extra_type(event)
+        bodyTable = event.find_all("table")[1].find("table")
+        if "CITY COUNCIL" in bodyTable.text:
+            return "City Council"
+        else:
+            return bodyTable.find_all("span")[3].text.title()
+
+    def get_event_minutes_item(
+        self,
+        event: Union[Tag, NavigableString, None],
+    ) -> List[ingestion_models.EventMinutesItem]:
+        """
+        Parse the page and gather the event minute items
+
+        Parameter:
+        ----------------
+        event: Union[Tag, NavigableString, None]
+            All elements in the page that we want to scrape
+
+        Returns:
+        ----------------
+        List[ingestion_models.EventMinutesItem]
+            All the event minute items gathered from the event on the page
+        """
+        log.info("start get items")
+        event_minutes_items = []
+        all_items = self.remove_extra_type(event).find_all("td", {"class": "style4"})
+        for item in all_items:
+            name = ""
+
+            for i in item.stripped_strings:
+                name = name + " " + repr(i).replace("'", "")
+
+            for a in item.find_all("a", href=True):
+                href = "https://houston.novusagenda.com/agendapublic/" + a["href"]
+
+            if name is not None and name != "":
+                event_minutes_items.append(
+                    ingestion_models.EventMinutesItem(
+                        minutes_item=ingestion_models.MinutesItem(name.strip()),
+                        supporting_files=href,
+                    )
+                )
+
+        return event_minutes_items
+
+    def get_diff_yearid(self, event_date: datetime) -> str:
+        """
+        Get the events in different years as the events for different
+        years are stored in different tabs. Can get multiple events
+        across years.
+
+        Parameters:
+        ---------------
+        event_date: datetime
+            The date of the event we are trying to parse
+
+        Returns:
+        ---------------
+        str
+            The year id that can locate the year tab where the event is stored
+        """
+        year = str(event_date.year)
+        year_id = "city-council-" + year
+        return year_id
+
+    def get_date_mainlink(self, element: Tag) -> str:
+        """
+        Find the main link for one event.
+
+        Parameters:
+        --------------
+        time: datetime
+            The date of one event
```

Comment:
This is still incorrect. This function takes `element` as a parameter, not `time`. Same problem for the docstring in `get_agenda()` below.

Reply:
Done